### ISSUE TITLE
Whether `flowmap` is suitable when the cam intrinsic parameters are known

### ISSUE THREAD

**Issue by Vincentqyw** [dcharatan/flowmap#10]
My application scenario involves using a standard `PINHOLE` camera cropped from a fisheye camera. Can `flowmap` fix the intrinsic parameters without optimization, solely used for estimating depth and pose information? Thx!

**Comment by dcharatan** [comment]
Yes, it's possible to run FlowMap with ground-truth/known intrinsics! If you modify the data loader to return the ground-truth intrinsics, you can append the following flag to the `python3 -m flowmap.overfit command` to tell FlowMap to use them:

```
model/intrinsics=ground_truth 
```

The intrinsics returned by the data loader should have shape `(frame, 3, 3)`, where each 3x3 matrix has the following format:

```
[[fx,  0, cx],
 [ 0, fy, cy],
 [ 0,  0,  1]]
```

Note that `fx`, `fy`, `cx`, and `cy` should be normalized, i.e., divided by image height and width. For example, `fx` is `fx in pixels / image width`.

If you don't want to deal with modifying the data loading, you could also just hard-code the intrinsics returned by `IntrinsicsGroundTruth` [here](https://github.com/dcharatan/flowmap/blob/a6f854fb9376fdaf9c40b2a0cf81a2378424fd84/flowmap/model/intrinsics/intrinsics_ground_truth.py).

**Comment by kk6398** [comment]
> Yes, it's possible to run FlowMap with ground-truth/known intrinsics! If you modify the data loader to return the ground-truth intrinsics, you can append the following flag to the `python3 -m flowmap.overfit command` to tell FlowMap to use them:
> 
> ```
> model/intrinsics=ground_truth 
> ```
> 
> The intrinsics returned by the data loader should have shape `(frame, 3, 3)`, where each 3x3 matrix has the following format:
> 
> ```
> [[fx,  0, cx],
>  [ 0, fy, cy],
>  [ 0,  0,  1]]
> ```
> 
> Note that `fx`, `fy`, `cx`, and `cy` should be normalized, i.e., divided by image height and width. For example, `fx` is `fx in pixels / image width`.
> 
> If you don't want to deal with modifying the data loading, you could also just hard-code the intrinsics returned by `IntrinsicsGroundTruth` [here](https://github.com/dcharatan/flowmap/blob/a6f854fb9376fdaf9c40b2a0cf81a2378424fd84/flowmap/model/intrinsics/intrinsics_ground_truth.py).

When I set the "model/intrinsics" as ground_truth, i.e. IntrinsicsGroundTruth, but returnd intrinsics is None. So How do I need to pass into the GT intrinsics? Can you provide some specific advice?
Looking forward your reply.

**Comment by dcharatan** [comment]
`InstrinsicsGroundTruth` just uses the intrinsics supplied by the data loader. To use it, you'll have to use a data loader that supplies intrinsics, like the [COLMAP one here](https://github.com/dcharatan/flowmap/blob/a6f854fb9376fdaf9c40b2a0cf81a2378424fd84/flowmap/dataset/dataset_colmap.py).

**Comment by kk6398** [comment]
Hi, 
I try InstrinsicsGroundTruth from the Colmap, and we use the GT intrinsic in every training iteration of model, i.e. `model_output = self.model(self.batch, self.flows, self.global_step)` in Line 53 of model_wrapper_overfit.py .  And     `intrinsics = self.intrinsics.forward(batch, flows, backbone_out, global_step)` in Line 71 of model.py, we return the GT intrinsic Colmap. So we only optimizer the extrinsic, depth(not including instrinsic), however, the final results is much worse than the original softmin or regressed intrinsic.

Can you tell me which step I make mistake?
Looking forward your reply sincerely.

**Comment by dcharatan** [comment]
Have you ensured that your ground-truth intrinsics are normalized (divided by image height/width)? Have you tried comparing the learned intrinsics to the ground-truth ones to see if they're similar?
